packages/react/accordion/src/Accordion.tsx

```diff
@@ -353,84 +365,88 @@ const AccordionHeader = React.forwardRef((props, forwardedRef) => {
 AccordionHeader.displayName = HEADER_NAME;
 
 /* -------------------------------------------------------------------------------------------------
- * AccordionButton
+ * AccordionTrigger
  * -----------------------------------------------------------------------------------------------*/
 
-const BUTTON_NAME = 'AccordionButton';
+const TRIGGER_NAME = 'AccordionTrigger';
 
-type AccordionButtonOwnProps = Polymorphic.OwnProps<typeof CollapsiblePrimitive.Button>;
-type AccordionButtonPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof CollapsiblePrimitive.Button>,
-  AccordionButtonOwnProps
+type AccordionTriggerOwnProps = Polymorphic.OwnProps<typeof CollapsiblePrimitive.Trigger>;
+type AccordionTriggerPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof CollapsiblePrimitive.Trigger>,
+  AccordionTriggerOwnProps
 >;
 
 /**
- * `AccordionButton` is the trigger that toggles the collapsed state of an `AccordionItem`. It
+ * `AccordionTrigger` is the trigger that toggles the collapsed state of an `AccordionItem`. It
  * should always be nested inside of an `AccordionHeader`.
  */
-const AccordionButton = React.forwardRef((props, forwardedRef) => {
-  const { buttonNodesRef } = useAccordionContext(BUTTON_NAME);
-  const itemContext = useAccordionItemContext(BUTTON_NAME);
+const AccordionTrigger = React.forwardRef((props, forwardedRef) => {
+  const { triggerNodesRef } = useAccordionContext(TRIGGER_NAME);
+  const valueContext = useAccordionValueContext(TRIGGER_NAME);
+  const itemContext = useAccordionItemContext(TRIGGER_NAME);
+  const collapsible = React.useContext(AccordionCollapsibleContext);
 
-  const ref = React.useRef<React.ElementRef<typeof CollapsiblePrimitive.Button>>(null);
+  const ref = React.useRef<React.ElementRef<typeof CollapsiblePrimitive.Trigger>>(null);
   const composedRefs = useComposedRefs(ref, forwardedRef);
 
   React.useEffect(() => {
-    const buttonNodes = buttonNodesRef.current;
-    const buttonNode = ref.current;
+    const triggerNodes = triggerNodesRef.current;
+    const triggerNode = ref.current;
 
-    if (buttonNode) {
-      buttonNodes.add(buttonNode);
+    if (triggerNode) {
+      triggerNodes.add(triggerNode);
       return () => {
-        buttonNodes.delete(buttonNode);
+        triggerNodes.delete(triggerNode);
       };
     }
     return;
-  }, [buttonNodesRef]);
+  }, [triggerNodesRef]);
 
   return (
-    <CollapsiblePrimitive.Button
-      aria-disabled={itemContext.open || undefined}
-      id={itemContext.buttonId}
+    <CollapsiblePrimitive.Trigger
+      aria-disabled={
+        (itemContext.open && valueContext.value.length === 1 && !collapsible) || undefined
```

Comment:
reading that context

Reply:
How about if we add the `AccordionCollapsibleContext.Provider` to both types and in the `AccordionSingle` its value will be the `collapsible` prop and in `AccordionMultiple` its value will always be `false`. 

This can then be simplified to `itemContext.open && !collapsible` so the trigger doesn't have implicit logic that assumes 1 value means single type.

Reply:
> How about if we add the AccordionCollapsibleContext.Provider to both types and in the AccordionSingle its value will be the collapsible prop and in AccordionMultiple its value will always be true.

Not sure we need to do it for multiple, I like that currently it's explicitly for single.

> This can then be simplified to itemContext.open && !collapsible so the trigger doesn't have implicit logic that assumes 1 value means single type.

Good catch, that was there because we were handling both before, but we're not anymore, so we can remove.

Reply:
I think we need it because otherwise `collapsible` will be undefined in the `AccordionMultiple` which is falsey. That's why I suggested doing `<AccordionCollapsibleContext.Provider value={true}>` in the `AccordionMultiple` component, because collapsible does exist for multiple, it's just always collapsible.

Reply:
ok, I don't have an issue with the falsey thing but I can see what you're saying, will do now.

Reply:
It's not about having a personal issue with the falsey thing, it's a bug is what I'm saying. If there is no `collapsible` context in `AccordionMultiple` then 

```
const collapsible = useCollapsibleContext() // = `undefined` which is == `false`;
```

So, where you have done `aria-disabled={context.open && !collapsible}` that would be `true` in `AccordionMultiple` which is wrong. It's a bug. That's why I'm saying we need to pass `collapsible` through context as `true`.